File: GrayMattr.Samples.Csharp/JsonStringify.cs
namespace GrayMattr.Samples.Csharp.JsonStringify;

public class JsonStringify
{
    static void Main()
    {
        GrayFile<IDictionary<string, string>> file1 = Mattr.Parse<IDictionary<string, string>>(
            string.Join("\n", ["---json", "{", "  \"name\": \"gray-matter\"", "}", "---", "This is content"])
        );

        file1.Stringify(
            new Dictionary<string, string>(),
            Mattr.DefaultOption<IDictionary<string, string>>().SetLanguage("yaml")
        );

        GrayFile<IDictionary<string, string>> file2 = Mattr.Parse<IDictionary<string, string>>(
            string.Join("\n", ["---", "title: Home", "---", "This is content"])
        );

        file2.Stringify(
            new Dictionary<string, string>(),
            Mattr.DefaultOption<IDictionary<string, string>>().SetLanguage("json")
        );
    }
}

File: GrayMattr.Samples/JsonStringify.fs
module JsonStringify

open GrayMattr

let file1: GrayFile<obj> =
    Mattr.Parse(
        [| "---json"
           "{"
           "  \"name\": \"gray-matter\""
           "}"
           "---"
           "This is content" |]
        |> String.concat "\n"
    )

file1.Stringify(dict [], Mattr.DefaultOption().SetLanguage("yaml")) |> ignore

let file2: GrayFile<obj> =
    Mattr.Parse([| "---"; "title: Home"; "---"; "This is content" |] |> String.concat "\n")

file2.Stringify(dict [], Mattr.DefaultOption().SetLanguage("json")) |> ignore
